specs/simple-serialize.md

```diff
@@ -97,6 +113,16 @@ fixed_parts = [part if part != None else variable_offsets[i] for i, part in enum
 return b"".join(fixed_parts + variable_parts)
 ```
 
+If `value` is an union type:
+
+Define value as an object that has properties `value.value` with the contained value, and `value.type_index` which indexes the type.
```

Comment:
IMO maybe too many "value"s in these sentences and code. Perhaps using `value.contained_value` would be more readable?

Reply:
Shall we use `object` instead? (That's what these things are after all, objects.)

Reply:
Feels wrong though to change the naming of the property though, it's really the first "value" that is only a placeholder.

Reply:
You mean `value.object` or `object.value` @JustinDrake ?

Reply:
I was thinking `object.value`

Reply:
I'm happy to go with that

Reply:
Oh but 'object' is a type in python :(